bazel/repository_locations.bzl

```diff
@@ -112,6 +112,18 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         release_date = "2022-04-04",
         cpe = "N/A",
     ),
+    com_github_aignas_rules_shellcheck = dict(
+        project_name = "Shellcheck rules for bazel",
+        project_desc = "Now you do not need to depend on the system shellcheck version in your bazel-managed (mono)repos.",
+        project_url = "https://github.com/aignas/rules_shellcheck",
+        version = "07582680053c4c9707e8ea6cf1d92e7bd2c3afcb",
+        sha256 = "a6ecf5437372d2137d8a203df586f955a863a1f4d6f0305369e0c75bc9d24160",
+        strip_prefix = "rules_shellcheck-{version}",
+        urls = ["https://github.com/aignas/rules_shellcheck/archive/{version}.tar.gz"],
+        release_date = "2022-05-24",
+        use_category = ["build"],
+        cpe = "N/A",
+    ),
```

Comment:
Poor OSS Scorecard score:
`scorecard --repo=https://github.com/aignas/rules_shellcheck`
```
RESULTS
-------
Aggregate score: 4.1 / 10

Check scores:
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
|  SCORE  |          NAME          |             REASON             |                             DOCUMENTATION/REMEDIATION                             |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 10 / 10 | Binary-Artifacts       | no binaries found in the repo  | https://github.com/ossf/scorecard/blob/main/docs/checks.md#binary-artifacts       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 3 / 10  | Branch-Protection      | branch protection is not       | https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection      |
|         |                        | maximal on development and all |                                                                                   |
|         |                        | release branches               |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 3 / 10  | CI-Tests               | 1 out of 3 merged PRs          | https://github.com/ossf/scorecard/blob/main/docs/checks.md#ci-tests               |
|         |                        | checked by a CI test -- score  |                                                                                   |
|         |                        | normalized to 3                |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 0 / 10  | CII-Best-Practices     | no badge detected              | https://github.com/ossf/scorecard/blob/main/docs/checks.md#cii-best-practices     |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 3 / 10  | Code-Review            | GitHub code reviews found for  | https://github.com/ossf/scorecard/blob/main/docs/checks.md#code-review            |
|         |                        | 3 commits out of the last 10   |                                                                                   |
|         |                        | -- score normalized to 3       |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 3 / 10  | Contributors           | 1 different organizations      | https://github.com/ossf/scorecard/blob/main/docs/checks.md#contributors           |
|         |                        | found -- score normalized to 3 |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 10 / 10 | Dangerous-Workflow     | no dangerous workflow patterns | https://github.com/ossf/scorecard/blob/main/docs/checks.md#dangerous-workflow     |
|         |                        | detected                       |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 0 / 10  | Dependency-Update-Tool | no update tool detected        | https://github.com/ossf/scorecard/blob/main/docs/checks.md#dependency-update-tool |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 0 / 10  | Fuzzing                | project is not fuzzed          | https://github.com/ossf/scorecard/blob/main/docs/checks.md#fuzzing                |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 10 / 10 | License                | license file detected          | https://github.com/ossf/scorecard/blob/main/docs/checks.md#license                |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 0 / 10  | Maintained             | 0 commit(s) out of 10 and 0    | https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained             |
|         |                        | issue activity out of 1 found  |                                                                                   |
|         |                        | in the last 90 days -- score   |                                                                                   |
|         |                        | normalized to 0                |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| ?       | Packaging              | no published package detected  | https://github.com/ossf/scorecard/blob/main/docs/checks.md#packaging              |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 8 / 10  | Pinned-Dependencies    | dependency not pinned by hash  | https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies    |
|         |                        | detected -- score normalized   |                                                                                   |
|         |                        | to 8                           |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 0 / 10  | SAST                   | SAST tool is not run on all    | https://github.com/ossf/scorecard/blob/main/docs/checks.md#sast                   |
|         |                        | commits -- score normalized to |                                                                                   |
|         |                        | 0                              |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 0 / 10  | Security-Policy        | security policy file not       | https://github.com/ossf/scorecard/blob/main/docs/checks.md#security-policy        |
|         |                        | detected                       |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| ?       | Signed-Releases        | no releases found              | https://github.com/ossf/scorecard/blob/main/docs/checks.md#signed-releases        |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 0 / 10  | Token-Permissions      | non read-only tokens detected  | https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions      |
|         |                        | in GitHub workflows            |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| 10 / 10 | Vulnerabilities        | no vulnerabilities detected    | https://github.com/ossf/scorecard/blob/main/docs/checks.md#vulnerabilities        |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
| ?       | Webhooks               | check is not supported for     | https://github.com/ossf/scorecard/blob/main/docs/checks.md#webhooks               |
|         |                        | this request: SCORECARD_V6     |                                                                                   |
|         |                        | is not set, not running the    |                                                                                   |
|         |                        | Webhook check                  |                                                                                   |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------|
 ```

Reply:
> Poor OSS Scorecard score:

yep, i was reluctant to add this as its not clear that its actively maintained/developed

what it does is pretty simple, and (almost) exactly what we need tho, so i guess the question is whether to just import the code

mho is use it for now but if necessary fork/import it in the future - cc @keith

Reply:
yea i think we could always just drop some rules in this repo directly if it became an issue

Reply:
These `rules_*` repositories are generally pretty bad from best practices perspective. I feel it would be consistent to just switch to this import, but I don't feel that strongly - one day we will want to start thinking about hardened build perhaps.

Reply:
hmm - i think that this is quite a useful and commonly needed bit of software - i would rather not move it into Envoy unless we have a pressing need

wrt this PR its based on @keith 's upstream PR so if we land it, it will be temporary while we resolve that 

there is no immediate rush to land this, its a nice-to-have and avoids a bit of dev pain, but my feeling is to land and iterate - but happy to update if that is preferred